app/lib/homework/shared/homework_tile_template.dart

```diff
@@ -70,3 +74,166 @@ class HomeworkTileTemplate extends StatelessWidget {
     );
   }
 }
+
+class _Subtitle extends StatelessWidget {
+  const _Subtitle({
+    required this.courseName,
+    required this.todoDate,
+    required this.todoDateColor,
+    required this.isCompleted,
+  });
+
+  final String courseName;
+  final String todoDate;
+  final Color? todoDateColor;
+  final bool isCompleted;
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.start,
+      children: [
+        _StrikeThrough(
+          isStrikeThrough: isCompleted,
+          delay: const Duration(milliseconds: 100),
+          child: Text(
+            courseName,
+            style: TextStyle(color: Colors.grey[600]),
+          ),
+        ),
+        const SizedBox(height: 4),
+        _StrikeThrough(
+          isStrikeThrough: isCompleted,
+          delay: const Duration(milliseconds: 200),
+          child: Text(todoDate, style: TextStyle(color: todoDateColor)),
+        ),
+      ],
+    );
+  }
+}
+
+class _Title extends StatelessWidget {
+  const _Title({
+    required this.title,
+    required this.isCompleted,
+  });
+
+  final String title;
+  final bool isCompleted;
+
+  @override
+  Widget build(BuildContext context) {
+    // Even though the row only contains one child, we use a row to make the
+    // title text ellipsis and not using the full width of the screen.
+    // Otherwise, the strike through would be drawn over the whole screen.
+    return Row(
+      children: [
+        Flexible(
+          child: _StrikeThrough(
+            isStrikeThrough: isCompleted,
+            child: Text(
+              title,
+              overflow: TextOverflow.ellipsis,
+              // We can only show one line of text in the title. Showing more
+              // lines would make the strike through look weird.
+              maxLines: 1,
```

Comment:
We need here to make a trade-off and reduce the `maxLines` from 2 to 1. I think it's okay since you normally know what the homework was when reading the first line. Otherwise, you just press on the homework and you get the full title.